File: simple-stack-example-basic-fragment/src/main/java/com/zhuinden/navigationexamplefrag/OtherKey.java
package com.zhuinden.navigationexamplefrag;

import com.google.auto.value.AutoValue;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class OtherKey extends BaseKey {
    public static OtherKey create() {
        return new AutoValue_OtherKey();
    }

    @Override
    protected BaseFragment createFragment() {
        return new OtherFragment();
    }
}

File: simple-stack-example-basic-kotlin/src/main/java/com/zhuinden/simplestackexamplekotlin/OtherKey.kt
package com.zhuinden.simplestackexamplekotlin

import android.annotation.SuppressLint
import kotlinx.android.parcel.Parcelize

/**
 * Created by Owner on 2017.11.13.
 */
@SuppressLint("ParcelCreator")
@Parcelize
data class OtherKey(val tag: String) : BaseKey() {
    constructor() : this("OtherKey")

    override fun createFragment(): BaseFragment = OtherFragment()
}
